lib/stat_tracker.rb

```diff
@@ -160,55 +160,47 @@ def lowest_scoring_home_team
   end
 
   # SEASON STATISTICS
-  # season stats helper methods -------------------
-  def season_games(season)
-    games.find_all { |game| game.season == season }
-  end
-  #-----------------------------
-
   def winningest_coach(season)
-    season_game_ids = season_games(season).map do |game|
-      game.game_id
-    end
-
-    season_game_teams = game_teams.find_all do |game|
-      season_game_ids.include?(game.game_id)
-    end
-
-    game_teams_by_coach = season_game_teams.group_by do |game_team|
-      game_team.head_coach
-    end
+    coach_win_percentage(season).max_by { |coach, record| record }[0]
+  end
 
-    coach_games_and_wins = Hash.new { |h,k| h[k] = Hash.new(0) }
+  def worst_coach(season)
+    coach_win_percentage(season).min_by { |coach, record| record }[0]
+  end
 
-    game_teams_by_coach.each do |coach, game_teams|
-      coach_games_and_wins[coach][:games] = game_teams.count
-      coach_games_and_wins[coach][:wins] = game_teams.find_all { |game| game.result == "WIN"}.count
-    end
+  def most_accurate_team(season)
+    most_accurate_team_id = team_accuracy(season).max_by { |team_id, acc| acc }[0]
+    teams.find { |team| team.team_id == most_accurate_team_id }.team_name
+  end
 
-    coach_win_percentage = {}
-    coach_games_and_wins.each do |coach, stats|
-      coach_win_percentage[coach] = stats[:wins].fdiv(stats[:games])
-    end
+  def least_accurate_team(season)
+    least_accurate_team_id = team_accuracy(season).min_by { |team_id, acc| acc }[0]
+    teams.find { |team| team.team_id == least_accurate_team_id }.team_name
+  end
 
-    coach_win_percentage.max_by { |coach, record| record }[0]
+  def most_tackles(season)
+    most_tackles_team_id = team_tackles(season).max_by { |team_id, tackles| tackles }[0]
+    teams.find { |team| team.team_id == most_tackles_team_id }.team_name
   end
 
-  def worst_coach(season)
-    season_game_ids = season_games(season).map do |game|
-      game.game_id
-    end
+  def fewest_tackles(season)
+    fewest_tackles_team_id = team_tackles(season).min_by { |team_id, tackles| tackles }[0]
+    teams.find { |team| team.team_id == fewest_tackles_team_id }.team_name
+  end
+  # season stats helper methods -------------------
+  def season_games(season)
+    games.find_all { |game| game.season == season }
+  end
 
-    season_game_teams = game_teams.find_all do |game|
-      season_game_ids.include?(game.game_id)
-    end
+  def season_game_teams(season)
+    season_game_ids = season_games(season).map { |game| game.game_id }
+    game_teams.find_all { |game| season_game_ids.include?(game.game_id) }
+  end
 
-    game_teams_by_coach = season_game_teams.group_by do |game_team|
-      game_team.head_coach
-    end
+  def coach_win_percentage(season)
+    game_teams_by_coach = season_game_teams(season).group_by { |game_team| game_team.head_coach }
```

Comment:
given SRP, this calculation should probably be separated into another helper method